src/flag_gems/ops/masked_fill.py

```diff
@@ -93,10 +67,4 @@ def masked_fill_(inp, mask, value):
     inp = inp.contiguous()
     mask = mask.contiguous()
     expand_mask = mask.expand(inp.shape)
-
-    N = inp.numel()
-    if N == 0:
-        return inp
-    grid = lambda meta: (triton.cdiv(N, meta["BLOCK_SIZE"]),)
-    masked_fill_kernel_self[grid](inp, expand_mask.to(torch.int), value, N)
-    return inp
+    return masked_fill_kernel(inp, expand_mask.to(torch.int), value, out0=inp)

```

Comment:
PointwiseDynamicFunction can handle non-contiguous inputs, so there is not need to make inp, mask contiguous beforehand.
Also, there is no need to cast mask to integer type, boolean is also fine.